tests/test_install.py

```diff
@@ -107,6 +111,8 @@ def test_symlink_other_python(self):
         assert_islink(self.tmpdir / 'site-packages2' / 'package1',
                       to=str(samples_dir / 'package1'))
         assert_isfile(self.tmpdir / 'scripts2' / 'pkg_script')
+        with (self.tmpdir / 'scripts2' / 'pkg_script').open() as f:
+            assert f.readline().strip() == "#!mock_python"
```

Comment:
This makes me think we should probably do an `abspath` on it. I don't think we want shebangs with a relative path.

Reply:
That should probably go very early in the code, just after argument parsing.
That probably involves searching in `PATH` too, because `flit install --python python` just works today as way to install in the current active virtualenv.

Reply:
A reliable (although a bit inefficient) possibility could be something like this

```python
def _normalize_python(python):
    if python == sys.executable:
        return python
    return subprocess.check_output([python, "-c", "import sys; print(sys.executable)"]).strip()
```

Reply:
possibly with a shortcut if `isabs(python)`

Reply:
In any case, one could argue that flit works better with this PR than not, if only for people providing a full path in `--python`. So maybe it could be merged as is? Alternatively, if you think the above idea about normalizing `args.python` is valid, I'm happy to add a commit with it to this PR.

Reply:
Yes, good points. In most cases I think `os.path.abspath(shutil.which(self.python))` should give the right answer, but it wouldn't surprise me if there's some odd corner cases with symlinks and things where that's different from running it in a subprocess and getting `sys.executable`.

I agree that this PR is a strict improvement, so I'm happy to merge this as if if you prefer.